selection/randomized/M_estimator.py

```diff
@@ -528,15 +528,18 @@ def new_log_density(query,
                             internal_state, 
                             opt_state):
 
-            full_state = reconstruct_full_from_internal(new_opt_transform, query.score_transform, internal_state, opt_state)
+            full_state = reconstruct_full_from_internal(new_opt_transform,
+                                                        query.score_transform,
+                                                        internal_state,
+                                                        opt_state)
             full_state = np.atleast_2d(full_state)
             p = query.penalty.shape[0]
             logdens = 0
 
             if inactive_marginal_groups.sum()>0:
                 full_state_plus = full_state + np.multiply(limits_marginal_groups, np.array(inactive_marginal_groups, np.float))
                 full_state_minus = full_state - np.multiply(limits_marginal_groups, np.array(inactive_marginal_groups, np.float))
-                logdens += np.log(_cdf(full_state_plus) - _cdf(full_state_minus)).sum()
+                logdens += np.log(_cdf(full_state_plus) - _cdf(full_state_minus))[:,inactive_marginal_groups].sum()
 
             logdens += log_dens(full_state[:,~inactive_marginal_groups])
             return np.squeeze(logdens) # should this be negative to match the gradient log density?
```

Comment:
I am confused in this function. logdens here is initialized as scalar but shouldn't it be a vector of size of the number of samples (so number of rows in opt_state). When we call new_log_density internal_sample and opt_sample are matrices. Also, I assume log_dens returns a vector of the size of a given vector in its argument?